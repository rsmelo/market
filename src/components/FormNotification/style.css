.formNotification {
  margin-top: 1em;
  display: block;
  padding: 1em;
}

.error {
  border: 1px solid var(--color-error-ligth);
  background: var(--color-error-ligther);
}

.success {
  border-left: 1px solid var(--color-success-light);
  background: var(--color-success-lighter);
}
