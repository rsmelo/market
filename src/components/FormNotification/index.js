import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import style from './style.css'

const FormNotification = ({ children, type }) => (
  <div className={classNames(style.formNotification, style[type])}>
    {children}
  </div>
)


FormNotification.defaultProps = {
  type: 'error',
}

FormNotification.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.string,
    PropTypes.node,
  ]).isRequired,
  type: PropTypes.oneOf(['success', 'error']),
}


export default FormNotification
