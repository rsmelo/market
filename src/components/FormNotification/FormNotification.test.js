import React from 'react'
import renderer from 'react-test-renderer'

import FormNotification from './index'

describe('FormNotification', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(<FormNotification>Test</FormNotification>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
