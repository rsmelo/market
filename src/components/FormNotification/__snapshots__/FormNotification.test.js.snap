// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormNotification should render correctly 1`] = `
<div
  className=""
>
  Test
</div>
`;
